pkg/serializer/internal/metrics/iterable_series.go

```diff
@@ -240,6 +242,27 @@ func (series *IterableSeries) MarshalSplitCompress(bufferContext *marshaler.Buff
 				}
 			}
 
+			if len(serie.Resources) > 0 {
+				for _, r := range serie.Resources {
+					err = ps.Embedded(seriesResources, func(ps *molecule.ProtoStream) error {
```

Comment:
What happens if device field is also present? (I know it may not be the case today, but if possible, it would be nice to avoid having a possibility to send broken payloads).

Reply:
I'm not sure which case you're thinking of but there are two that come to mind for me:
* Device is set on the serie and we also have Resources set. That's the case we were already covering in the tests and the result is that device is set as a resource *and* the resources are also set as resources.
* Device is set on the serie and we also have a device resource with something like `dd.internal.resource:device:something`. In that scenario, the emitted payload would have _both_ device resources emitted. Without knowing more about the device resource type, I can't tell if that's okay but it seems fine at a high-level to be sending different resources of the same type.

Reply:
Ah, I missed that `resources` is a repeated field and it is expected that we serialize it multiple times. No problem here then.